### PR TITLE
fix: 🐛 change end time for day and week view #298

### LINKED ISSUE #298: Can i change start time and end time in DayView?
Screenshot 2023-11-27 at 11 44 13 AM
By the default, the time range of the DayView from 00:00 -> 24:00, so can i change to 08:00 -> 20:00?
Thank you so much!

### PR DESCRIPTION
**Pull request by apurva010**
- Added end time support for day and weeek view to set end time of day and week.

# Description

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #298 

[Contributor Guide]: https://github.com/SimformSolutionsPvtLtd/flutter_calendar_view/blob/master/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: lib/src/day_view/day_view.dart [hunk 1]
```diff
@@ -336,7 +341,10 @@ class DayViewState<T extends Object?> extends State<DayView<T>> {
     super.initState();
 
     _startHour = widget.startHour ?? 0;
+    _endHour = widget.endHour ?? 24;
```
Comment (vatsaltanna-simformsolutions) [inline lib/src/day_view/day_view.dart:344]: use constant file value here

File: lib/src/week_view/week_view.dart [hunk 2]
```diff
@@ -353,8 +358,10 @@ class WeekViewState<T extends Object?> extends State<WeekView<T>> {
     super.initState();
 
     _startHour = widget.startHour ?? 0;
+    _endHour = widget.endHour ?? 24;
```
Comment (vatsaltanna-simformsolutions) [inline lib/src/week_view/week_view.dart:361]: ^^

File: lib/src/week_view/week_view.dart [hunk 3]
```diff
@@ -353,8 +358,10 @@ class WeekViewState<T extends Object?> extends State<WeekView<T>> {
...
     //Security to avoid any height bug
     if (_startHour > 24) _startHour = 0;
+    if (_endHour >= 24 || _startHour > _endHour) _endHour = 24;
```
Comment (vatsaltanna-simformsolutions) [inline lib/src/week_view/week_view.dart:364]: ^^

File: README.md [hunk 4]
```diff
@@ -179,7 +179,8 @@ DayView(
     onEventDoubleTap: (events, date) => print(events),
     onEventLongTap: (events, date) => print(events),
     onDateLongPress: (date) => print(date),
-    startHour: 5 // To set the first hour displayed (ex: 05:00)
+    startHour: 51 // To set the first hour displayed (ex: 05:00)
```
Comment (PRBaraiya) [inline README.md:182]: Make this 5.

File: lib/src/components/_internal_components.dart [hunk 5]
```diff
@@ -162,6 +169,7 @@ class TimeLine extends StatefulWidget {
     this.showHalfHours = false,
     this.showQuarterHours = false,
     required this.liveTimeIndicatorSettings,
+    required this.endHour,
```
Comment (PRBaraiya) [inline lib/src/components/_internal_components.dart:172]: This should not be required. set default values for start and end hour.

Comment (PRBaraiya) [reply lib/src/components/_internal_components.dart:172]: Also, add an assert condition to check start hour is not >= then end hour.

File: lib/src/day_view/day_view.dart [hunk 6]
```diff
@@ -223,6 +223,9 @@ class DayView<T extends Object?> extends StatefulWidget {
   /// Emulate vertical line offset from hour line starts.
   final double emulateVerticalOffsetBy;
 
+  /// This field will be used to set end hour for day view
+  final int? endHour;
```
Comment (PRBaraiya) [inline lib/src/day_view/day_view.dart:227]: No need to make this nullable. Set default value to 24 hrs.

Comment (PRBaraiya) [reply lib/src/day_view/day_view.dart:227]: Make same changes for start hour as well.

File: lib/src/painters.dart [hunk 7]
```diff
@@ -53,6 +56,7 @@ class HourLinePainter extends CustomPainter {
     required this.showVerticalLine,
     required this.startHour,
     required this.emulateVerticalOffsetBy,
+    required this.endHour,
```
Comment (PRBaraiya) [inline lib/src/painters.dart:59]: remove required and set default value.

**Review by PRBaraiya — CHANGES_REQUESTED** [review]

**Review by vatsaltanna-simformsolutions — APPROVED** [review]

**Pull request merged**